mod/editpost.php

```diff
@@ -95,7 +95,7 @@ function editpost_content(App $a)
 
 	$o .= replace_macros($tpl, [
 		'$is_edit' => true,
-		'$return_path' => $_SESSION['return_url'],
+		'$return_path' => $_SERVER['HTTP_REFERER'] ,
```

Comment:
Using the Referer is a bad practice, many browsers entirely scrub the Header from requests. Instead, the edit post form/page should have a "return_path" element that's manually set to the originating URL. In the `editpost_content` case, this means adding a new GET parameter filled with the calling page with its own URL.